app/grandchallenge/reader_studies/migrations/0008_auto_20210226_0949.py

```diff
@@ -1,17 +1,60 @@
-# Generated by Django 3.1.6 on 2021-02-24 15:03
+# Generated by Django 3.1.6 on 2021-02-26 09:49
 
 from django.db import migrations, models
 
 import grandchallenge.core.validators
 
 
+IMAGE_PORT_EMPTY_PLACEHOLDER = 9
+
+
+def remap_image_ports(apps, _):
+    question = apps.get_model("reader_studies", "Question")
+    # Migration does not pick up "" and sets this to NULL, so using a placeholder here.
+    remap = {"": IMAGE_PORT_EMPTY_PLACEHOLDER, "M": "0", "S": "1"}
+    for question in question.objects.all():
+        question.image_port = remap[question.image_port]
+        question.save()
+
+
+def clear_placeholder(apps, _):
+    question = apps.get_model("reader_studies", "Question")
+    for question in question.objects.filter(
+        image_port=IMAGE_PORT_EMPTY_PLACEHOLDER
+    ):
+        question.image_port = None
+        question.save()
+
+
 class Migration(migrations.Migration):
 
     dependencies = [
         ("reader_studies", "0007_auto_20210201_2220"),
     ]
 
     operations = [
+        migrations.RunPython(remap_image_ports),
```

Comment:
If you do use `RunPython` you are missing `elidable=True` as nothing needs to be done with an empty table.

Reply:
Going to revert to a minimal migration, but the `elidable=True` is a valuable item I will remember. Thanks!